src/renderer.zig

```diff
@@ -606,6 +610,312 @@ pub const MenuBackGround = struct {
+		.{0.022700000000, 0.008210000000, 0.000000000000},
+		.{0.015840000000, 0.005723000000, 0.000000000000},
+		.{0.011359160000, 0.004102000000, 0.000000000000},
+		.{0.008110916000, 0.002929000000, 0.000000000000},
+		.{0.005790346000, 0.002091000000, 0.000000000000},
+		.{0.004109457000, 0.001484000000, 0.000000000000},
+		.{0.002899327000, 0.001047000000, 0.000000000000},
+		.{0.002049190000, 0.000740000000, 0.000000000000},
+		.{0.001439971000, 0.000520000000, 0.000000000000},
+		.{0.000999949300, 0.000361100000, 0.000000000000},
+		.{0.000690078600, 0.000249200000, 0.000000000000},
+		.{0.000476021300, 0.000171900000, 0.000000000000},
+		.{0.000332301100, 0.000120000000, 0.000000000000},
+		.{0.000234826100, 0.000084800000, 0.000000000000},
+		.{0.000166150500, 0.000060000000, 0.000000000000},
+		.{0.000117413000, 0.000042400000, 0.000000000000},
+		.{0.000083075270, 0.000030000000, 0.000000000000},
+		.{0.000058706520, 0.000021200000, 0.000000000000},
+		.{0.000041509940, 0.000014990000, 0.000000000000},
+		.{0.000029353260, 0.000010600000, 0.000000000000},
+		.{0.000020673830, 0.000007465700, 0.000000000000},
+		.{0.000014559770, 0.000005257800, 0.000000000000},
+		.{0.000010253980, 0.000003702900, 0.000000000000},
+		.{0.000007221456, 0.000002607800, 0.000000000000},
+		.{0.000005085868, 0.000001836600, 0.000000000000},
+		.{0.000003581652, 0.000001293400, 0.000000000000},
+		.{0.000002522525, 0.000000910930, 0.000000000000},
+		.{0.000001776509, 0.000000641530, 0.000000000000},
+		.{0.000001251141, 0.000000451810, 0.000000000000}
+	};
+
+	fn plankRadiance(temperature: f64, wavelength: f64) f64 {
+		const meters = wavelength * 1e-9;
+		const c2 = speedOfLight * speedOfLight;
+		const meter5 = meters * meters * meters * meters * meters;
+		return (2 * planck * c2) / (meter5 * (@exp(planck * speedOfLight / (meters * boltzmann * temperature)) - 1));
+	}
+
+	fn getStarColor(temperature: f64, magnitude: f64) Vec3d {
+		var total: Vec3d = @splat(0);
+		for (0..wavelengths.len) |i| {
+			const wavelength: f64 = @floatFromInt(i * 5 + 360);
+			const spectrum = plankRadiance(temperature, wavelength);
+			total += @as(Vec3d, @splat(spectrum * 5)) * wavelengths[i];
+		}
+
+		total /= @splat(@reduce(.Max, total));
+
+		const conv = [_]Vec3d{
+			.{ 3.2406, -0.9689, 0.0557 },
+			.{ -1.5372, 1.8758, -0.2040 },
+			.{ -0.4986, 0.0415, 1.0570 },
+		};
+
+		var rgb = conv[0] * @as(Vec3d, @splat(total[0])) + conv[1] * @as(Vec3d, @splat(total[1])) + conv[2] * @as(Vec3d, @splat(total[2]));
+
+		const threshold = 0.0031308;
+		const a = @as(Vec3d, @splat(1.055));
+		const b = @as(Vec3d, @splat(12.92));
+		const gamma = 1.0 / 2.4;
+
+		const mask = rgb > @as(Vec3d, @splat(threshold));
+		const corrected = a * Vec3d {
+			std.math.pow(f64, rgb[0], gamma),
+			std.math.pow(f64, rgb[1], gamma),
+			std.math.pow(f64, rgb[2], gamma),
+		} - @as(Vec3d, @splat(0.055));
+		rgb = @select(f64, mask, corrected, rgb * b);
```

Comment:
Instead of doing all these complex calculations, I think it would be best to just read the temperature→color conversion map from a texture instead.
This would allow us to use physically inaccurate colors (green, purple) and might be nice for texture packs and events too (imagine a terraria-like blood-moon with red stars).